voter-dapp/src/ActiveRequests.js

```diff
@@ -7,19 +7,69 @@ import TableHead from "@material-ui/core/TableHead";
 import TableRow from "@material-ui/core/TableRow";
 
 import { formatDate } from "./common/FormattingUtils.js";
+import { VotePhasesEnum } from "./common/Enums.js";
 
 function ActiveRequests() {
   const { drizzle, useCacheCall } = drizzleReactHooks.useDrizzle();
+  const { web3 } = drizzle;
+
   const pendingRequests = useCacheCall("Voting", "getPendingRequests");
-  if (!pendingRequests) {
+  const currentRoundId = useCacheCall("Voting", "getCurrentRoundId");
+  const votePhase = useCacheCall("Voting", "getVotePhase");
+  const account = drizzleReactHooks.useDrizzleState(drizzleState => ({
+    account: drizzleState.accounts[0]
+  })).account;
+  const initialFetchComplete = pendingRequests && currentRoundId && votePhase && account;
+
+  const voteStatuses = useCacheCall(["Voting"], call => {
+    if (!initialFetchComplete) {
+      return null;
+    }
+    return pendingRequests.map(request => ({
+      committedValue: call(
+        "Voting",
+        "getMessage",
+        account,
+        web3.utils.soliditySha3(request.identifier, request.time, currentRoundId)
+      ),
+      // Technically this call can be avoided if `votePhase == COMMIT`.
+      hasRevealed: call("Voting", "hasRevealedVote", request.identifier, request.time)
+    }));
+  });
+  const subsequentFetchComplete =
+    Boolean(voteStatuses) &&
+    // Each of the subfetches has to complete. In drizzle, `undefined` means incomplete, while `null` means complete
+    // but the fetched value was null, e.g., no `comittedValue` existed.
+    voteStatuses.every(
+      voteStatus => typeof voteStatus.committedValue !== "undefined" && typeof voteStatus.hasRevealed !== "undefined"
+    );
+
+  if (!initialFetchComplete || !subsequentFetchComplete) {
```

Comment:
Just curious, since we're conditionally returning early here, we can't make any react hook calls after this, right? If you did decide to make a react hook call after it, would react detect that and blow up immediately? Or would the error be more subtle?

Reply:
That's right, we can't make hook calls after this point. The compiler will find some cases (the compiler will return a compilation error in *this* case), but probably not all cases.